x-pack/plugins/observability_solution/logs_explorer/public/controller/create_controller.ts

```diff
@@ -68,6 +70,7 @@ export const createLogsExplorerControllerFactory =
 
     const machine = createLogsExplorerControllerStateMachine({
       datasetsClient,
+      plugins,
       initialContext,
       query: discoverServices.data.query,
       toasts: core.notifications.toasts,
```

Comment:
Nit: the argument list is growing a bit inconsistent. By forwarding the `plugins` here a future reader might overlook that some specific plugins must be overridden (such as `data`). They might therefore be tempted to access `data.query` inside of `createLogsExplorerControllerStateMachine` via the second argument and not the fourth argument. Can we somehow reduce that risk by either (a) passing all separately or (b) passing all in one object?